metabulo/views.py

```diff
@@ -325,9 +325,12 @@ def _apply_transforms(csv_file):
     return table
 
 
-def _get_pca_data(csv_file):
+def _get_pca_data(csv_file, max_components=2):
     table = _apply_transforms(csv_file)
-    max_components = int(request.args.get('max_components', 2))
+
+    max_components_arg = request.args.get('max_components')
+    if max_components_arg is not None:
+        max_components = int(max_components_arg)
```

Comment:
The one-liner in the existing code:

```
max_components = int(request.args.get('max_components', 2))
```

seems to do everything the changed code does. Since no one is using the function keyword argument, why was it added?

Reply:
One of the commits in the sequence did add a keyword argument to this function, to allow me to specify `max_components` from an internal invocation of the function (when usually you have to send the value in through the request args).

I will fix this on this branch of course, but I think a better solution here is to use webargs to inject query param arguments into the flask routes themselves, then make sure these internal functions simply have keyword args with valid defaults rather than asking for the request args from them.

Reply:
See 4f7a359c6b13b8263a8d4a510268dac404153ce6 and #170.